Meta/Lagom/Tools/CodeGenerators/LibUnicode/GenerateUnicodeData.cpp

```diff
@@ -1724,42 +1724,44 @@ struct PropertyMetadata {
 // this process reduces over 1 million entries (0x10ffff) to ~44,030.
 //
 // For much more in-depth reading, see: https://icu.unicode.org/design/struct/utrie
-static ErrorOr<void> create_code_point_tables(UnicodeData& unicode_data)
+static constexpr auto MAX_CODE_POINT = 0x10ffffu;
+
+template<typename T>
+static ErrorOr<void> update_tables(u32 code_point, CodePointTables<T>& tables, auto& metadata, auto const& values)
 {
-    static constexpr auto MAX_CODE_POINT = 0x10ffffu;
+    static constexpr auto BLOCK_SIZE = CODE_POINT_TABLES_LSB_MASK + 1;
+
+    size_t unique_properties_index = 0;
+    if (auto block_index = tables.unique_properties.find_first_index(values); block_index.has_value()) {
+        unique_properties_index = *block_index;
+    } else {
+        unique_properties_index = tables.unique_properties.size();
+        TRY(tables.unique_properties.try_append(values));
+    }
 
-    auto update_tables = [&](auto code_point, auto& tables, auto& metadata, auto const& values) -> ErrorOr<void> {
```

Comment:
I'm sure I'd ask the same thing about this if I had the energy to read the diff.